File: 1063-best-sightseeing-pair/best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        n = len(values)
        mx = 0
        dp = [0] * n
        dp[0] = values[0]

        for i in range(1, n):
            dp[i] = max(dp[i - 1], values[i - 1] + i - 1)
            mx = max(mx, dp[i] + values[i] - i)

        return mx

File: 1063-best-sightseeing-pair/best-sightseeing-pair.js
/**
 * @param {number[]} values
 * @return {number}
 */
var maxScoreSightseeingPair = function (values) {
    const n = values.length;
    let mx = 0, cur = 0;

    for (let i = 1; i < n; i++) {
        cur = Math.max(cur, values[i - 1] + i - 1);
        mx = Math.max(mx, cur + values[i] - i);
    }

    return mx;
};